cmake/DynamicVersion.cmake

```diff
@@ -439,6 +452,23 @@ function(get_dynamic_version)
 		string(JSON data SET
 				${data} commit \"${git-hash}\")
 		file(WRITE ${ARGS_TMP_FOLDER}/.git_commit ${git-hash})
+        # Get full describe with distance
+        execute_process(COMMAND ${GIT_EXECUTABLE} describe --tags --long --match=?[0-9.]*
```

Comment:
Can you whow me a quick output of this when distance is 0 vs not? Iwonder if we can move this to the top one

Reply:
beyond tag
```
> git describe --tags
v1.9-5-g81314f83d
> git describe --tags --long
v1.9-5-g81314f83d
```

on-tag
```
> git describe --tags
v0.28.1
> git describe --tags --long
v0.28.1-0-gcbec059
```

yes, the two calls could probably be merged.

Reply:
The stored contents of `.git_describe` will change when I merge the calls -- hopefully that's ok. Here's a sample contents of the json file.

```
{
  "allow-fails" : false,
  "commit" : "82c98d38963f6b4a5718a0330de60d6e442152a5",
  "describe" : "v2.8.0-37-g82c98d38",
  "distance" : "37",
  "failed" : false,
  "short_sha" : "82c98d38",
  "version" : "2.8.0"
}
```

and currently stored files:
```
> ls -a1 buildcm/tmp/
.
..
.DynamicVersion.json
.git_commit
.git_describe
.version
```

Reply:
Yes, that's perfect